SwiftReflector/SwiftXmlReflection/ParameterItem.cs

```diff
@@ -165,6 +169,15 @@ public static bool AreEqualIgnoreNames (IList<ParameterItem> pl1, IList<Paramete
 			return true;
 		}
 
+		static ParameterItem SubstituteSelfFromParent (FunctionDeclaration func, ParameterItem p)
+		{
+			if (func.Parent == null || !p.TypeSpec.HasDynamicSelf)
+				return p;
+			p = new ParameterItem (p);
+			p.TypeSpec = p.TypeSpec.ReplaceName ("Self", func.Parent.ToFullyQualifiedNameWithGenerics ());
```

Comment:
you just re-assigned `p` so is `p.TypeSpec` what you want (the new one vs the one given as an argument) ?

Reply:
Yes. We run it through a copy constructor to not alter the input and then side-effect the result.